src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/VisualStudioLanguageServerFeatureOptions.cs

```diff
@@ -30,67 +22,30 @@ internal class VisualStudioLanguageServerFeatureOptions : LanguageServerFeatureO
+        _featureFlagService = featureFlagService;
         _lspEditorFeatureDetector = lspEditorFeatureDetector;
 
-        _showAllCSharpCodeActions = new Lazy<bool>(() =>
-        {
-            var featureFlags = (IVsFeatureFlags)Package.GetGlobalService(typeof(SVsFeatureFlags));
-            var showAllCSharpCodeActions = featureFlags.IsFeatureEnabled(ShowAllCSharpCodeActionsFeatureFlag, defaultValue: false);
-            return showAllCSharpCodeActions;
-        });
-
-        _includeProjectKeyInGeneratedFilePath = new Lazy<bool>(() =>
-        {
-            var featureFlags = (IVsFeatureFlags)Package.GetGlobalService(typeof(SVsFeatureFlags));
-            var includeProjectKeyInGeneratedFilePath = featureFlags.IsFeatureEnabled(IncludeProjectKeyInGeneratedFilePathFeatureFlag, defaultValue: true);
-            return includeProjectKeyInGeneratedFilePath;
-        });
-
-        _usePreciseSemanticTokenRanges = new Lazy<bool>(() =>
-        {
-            var featureFlags = (IVsFeatureFlags)Package.GetGlobalService(typeof(SVsFeatureFlags));
-            var usePreciseSemanticTokenRanges = featureFlags.IsFeatureEnabled(UsePreciseSemanticTokenRangesFeatureFlag, defaultValue: false);
-            return usePreciseSemanticTokenRanges;
-        });
-
-        _useRazorCohostServer = new Lazy<bool>(() =>
-        {
-            var featureFlags = (IVsFeatureFlags)Package.GetGlobalService(typeof(SVsFeatureFlags));
-            var useRazorCohostServer = featureFlags.IsFeatureEnabled(UseRazorCohostServerFeatureFlag, defaultValue: false);
-            return useRazorCohostServer;
-        });
-
-        _disableRazorLanguageServer = new Lazy<bool>(() =>
-        {
-            var featureFlags = (IVsFeatureFlags)Package.GetGlobalService(typeof(SVsFeatureFlags));
-            var disableRazorLanguageServer = featureFlags.IsFeatureEnabled(DisableRazorLanguageServerFeatureFlag, defaultValue: false);
-            return disableRazorLanguageServer;
-        });
-
-        _forceRuntimeCodeGeneration = new Lazy<bool>(() =>
-        {
-            var featureFlags = (IVsFeatureFlags)Package.GetGlobalService(typeof(SVsFeatureFlags));
-            var forceRuntimeCodeGeneration = featureFlags.IsFeatureEnabled(ForceRuntimeCodeGenerationFeatureFlag, defaultValue: false);
-            return forceRuntimeCodeGeneration;
-        });
-
-        _useProjectConfigurationEndpoint = new Lazy<bool>(() =>
-        {
-            var featureFlags = (IVsFeatureFlags)Package.GetGlobalService(typeof(SVsFeatureFlags));
-            var useProjectConfigurationEndpoint = featureFlags.IsFeatureEnabled(UseProjectConfigurationEndpointFeatureFlag, defaultValue: false);
-            return useProjectConfigurationEndpoint;
-        });
+        _showAllCSharpCodeActions = IsFeatureEnabledOrFalse(WellKnownFeatureFlagNames.ShowAllCSharpCodeActions);
+        _includeProjectKeyInGeneratedFilePath = IsFeatureEnabledOrTrue(WellKnownFeatureFlagNames.IncludeProjectKeyInGeneratedFilePath);
```

Comment:
Nit: To make this easier to parse, can we split things into two sections with comment headers like "On by default" and "Off by default"?

Reply:
Good suggestion